### DISCUSSION TITLE
RAG meta info and cost control

### DISCUSSION THREAD

**Discussion by PhilippChEnBW** [glowingjade/obsidian-smart-composer#93] [Ideas & Feature Requests]
Hi there!
First of all, this seems like an amazing product! I've tried it out summarizing specific pages and my whole vault and was blown away by how seamlessly the whole experience is integrated.

The thing I'm currently missing the most would be a method to get some insight into which info is sent to the LLM. All I could see was that summarizing my whole vault took up around 10 cents (not that many documents in there) and about 14k tokens for input+output combined. I'd love to see which data was actually sent to the LLM - ideally from within Obsidian. Also, it'd be great to be able to set a maximum number of chunks that should be selected for forwarding to the LLM to keep cost under control.

Best regards
Philipp

**Comment by glowingjade** [comment]
Hi! Thank you for using our plugin.

To enhance clarity on the data sent to the LLM, we have two items in our backlog:

- [Feature: Display RAG results](https://github.com/users/glowingjade/projects/1/views/3?pane=issue&itemId=84774587)
- [Feature: Show token usage statistics for each message](https://github.com/users/glowingjade/projects/1/views/3?pane=issue&itemId=84970310)

Do you think this will address your issue? We're planning to implement these changes soon!

Additionally, you can set a token threshold in the RAG settings. If file contents exceed this limit, RAG trims contexts accordingly. You can also configure how many chunks are included in the context through the RAG limit settings. More details can be found here: [1.4 Advanced Configuration#RAG-settings](https://github.com/glowingjade/obsidian-smart-composer/wiki/1.4-Advanced-Configuration#rag-settings)

**Reply by PhilippChEnBW** [reply]
Thanks a lot, these look exactly like what I'm looking for :)

**Reply by glowingjade** [reply]
Thanks for clarifying! We'll implement these features soon. Stay tuned!

**Reply by glowingjade** [reply]
These two features have been released! I'll close this discussion.
